tests/rptest/utils/si_utils.py

```diff
@@ -489,9 +489,20 @@ def kafka_start_offset(manifest) -> Optional[int]:
             return None
 
         start_model_offset = manifest['start_offset']
-        first_segment = min(manifest['segments'].values(),
-                            key=lambda seg: seg['base_offset'])
-        delta = first_segment['delta_offset']
+        segments_at_offset = list(
+            filter(lambda seg: seg['base_offset'] == start_model_offset,
+                   manifest['segments'].values()))
+        delta = -1
+        if len(segments_at_offset) == 0:
+            # If there's no segment that matches exactly, just use the first
+            # segment's delta, provided it's ahead of the start.
+            first_segment = min(manifest['segments'].values(),
+                                key=lambda seg: seg['base_offset'])
+            if first_segment['base_offset'] < start_model_offset:
+                return None
+            delta = first_segment['delta_offset']
+        else:
+            delta = segments_at_offset[0]['delta_offset']
 
         return start_model_offset - delta
```

Comment:
The case where retention advances the `start_offset` to `commited_offset + 1` of the last segment at the time is not covered.

Reply:
That's interesting, I was just copying what I saw in the partition manifest code.

Reply:
https://github.com/redpanda-data/redpanda/blob/e133de2fe76ae34d8e9972eb8a225dd346d3628a/src/v/cloud_storage/partition_manifest.cc#L308

Do you have a pointer to what Redpanda does for `committed_offset + 1`?

Reply:
https://github.com/redpanda-data/redpanda/blob/1e13eab0b168fdf05c177110d8110f62abbea1f9/src/v/archival/retention_calculator.cc#L113-L117